raiden-ts/src/transfers/epics/mediate.ts

```diff
@@ -34,34 +79,51 @@ function shouldMediate(action: transferSigned, address: Address, { caps }: Raide
  * @param deps - Raiden epic deps
  * @param deps.address - Our address
  * @param deps.config$ - Latest observable
+ * @param deps.log - Logger instance
  * @returns Observable of outbound transfer.request actions
  */
 export function transferMediateEpic(
   action$: Observable<RaidenAction>,
-  {}: Observable<RaidenState>,
-  { address, config$ }: RaidenEpicDeps,
+  state$: Observable<RaidenState>,
+  { address, config$, log }: RaidenEpicDeps,
 ) {
   return action$.pipe(
     filter(transferSigned.is),
-    withLatestFrom(config$),
+    withLatestFrom(config$, state$),
     filter(([action, config]) => shouldMediate(action, address, config)),
-    map(([{ payload: { message: transf }, meta: { secrethash } }]) =>
+    map(([action, config, state]) => {
+      const transferState = state.transfers[transferKey(action.meta)];
+      const locked = action.payload.message;
+
+      const tokenNetwork = locked.token_network_address;
+      const secrethash = action.meta.secrethash;
+      const receivedPartner = transferState.partner;
+
+      const routes = filterRoutes(locked.metadata.routes, state, tokenNetwork, { address, log });
+      if (!routes) return;
+      const sendPartner = routes[0].route[0];
+
+      const channelIn = state.channels[channelKey({ tokenNetwork, partner: receivedPartner })];
+      const channelOut = state.channels[channelKey({ tokenNetwork, partner: sendPartner })];
+
+      const fee = calculateMediationFee(locked.lock.amount, channelIn, channelOut, { config });
+
       // request an outbound transfer to target; will fail if already sent
-      transfer.request(
+      return transfer.request(
         {
-          tokenNetwork: transf.token_network_address,
-          target: transf.target,
-          value: transf.lock.amount, // TODO: implement fees
-          paymentId: transf.payment_identifier,
-          paths: transf.metadata.routes.map(({ route }) => ({
-            path: route.slice(1), // TODO: validate we were the first/next hop
-            fee: Zero as Int<32>, // TODO: implement fees
-          })),
-          expiration: transf.lock.expiration.toNumber(),
-          initiator: transf.initiator,
+          tokenNetwork: locked.token_network_address,
+          target: locked.target,
+          paymentId: locked.payment_identifier,
+          // on a transfer.request, fee is added to the value to get final sent amount, therefore
+          // here it needs to contain a negative fee, which we will "take" instead of pay
```

Comment:
All comments are related to the line or block below. This is just an implementation detail, it's no issue, this just mean that in the code which will handle this `transfer.request` (intellisense you led you there), the `fee` param will be added to the `value`, therefore, since on mediation we want to **take** fees instead of **pay** them (default send usacase), the `fee` needs to be negative here, and this is what this comment is telling you.